docs/resources/repository_file.md

```diff
@@ -0,0 +1,53 @@
+# gitlab\_repository\_file
+
+This resource allows you to create and manage GitLab repository files.
+
+## Example Usage
+
+```hcl
+resource "gitlab_group" "this" {
+    name        = "example"
+    path        = "example"
+    description = "An example group"
+}
+resource "gitlab_project" "this" {
+    name                   = "example"
+    namespace_id           = gitlab_group.this.id
+    initialize_with_readme = true
+}
+resource "gitlab_repository_file" "this" {
+    project        = gitlab_project.this.id
+    file_path      = "meow.txt"
+    branch         = "main"
+    content        = base64encode("Meow goes the cat")
+    author_email   = "terraform@example.com"
+    author_name    = "Terraform"
+    commit_message = "feature: add meow file"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `project` - (Required) The ID of the project.
+
+* `file_path` - (Required) The full path of the file.
+
+* `branch` - (Required) Name of the branch to which to commit to.
+
+* `start_branch` - (Required) Name of the branch to start the new commit from.
```

Comment:
According to: https://docs.gitlab.com/ee/api/commits.html start branch is not required argument

Reply:
good catch. Updated 🎉